internal/host/validator.go

```diff
@@ -142,30 +155,72 @@ func (c *validationContext) loadGeneralMinRequirements(hwValidator hardware.Vali
 	return err
 }
 
+func (c *validationContext) loadGeneralInfraEnvMinRequirements(hwValidator hardware.Validator) error {
+	requirements, err := hwValidator.GetPreflightInfraEnvHardwareRequirements(context.TODO(), c.infraEnv)
+	if err != nil {
+		return err
+	}
+
+	c.minCPUCoresRequirement = int64(math.Min(float64(requirements.Ocp.Master.Quantitative.CPUCores), float64(requirements.Ocp.Worker.Quantitative.CPUCores)))
+	c.minRAMMibRequirement = int64(math.Min(float64(requirements.Ocp.Master.Quantitative.RAMMib), float64(requirements.Ocp.Worker.Quantitative.RAMMib)))
+
+	return err
+}
+
 func newValidationContext(host *models.Host, c *common.Cluster, db *gorm.DB, hwValidator hardware.Validator) (*validationContext, error) {
 	ret := &validationContext{
-		host:    host,
-		db:      db,
-		cluster: c,
+		host:     host,
+		db:       db,
+		cluster:  c,
+		infraEnv: nil,
 	}
-	err := ret.loadCluster()
-	if err == nil {
+	if host.ClusterID != "" {
```

Comment:
newValidationContext is called with cluster parameter.  Can the ClusterID be empty in this case?

Reply:
it is called with host and cluster. cluster might be nil, even if it exists. in case cluster is not passed, it will be loaded by the loadCluster function. host.ClusterIDF condition differentiates between day1 cluster hosts and infra-env (pool) hosts

Reply:
But where is it called from (with InfraEnv only) ?

Reply:
it will be called during validation for infra-env host statemachine. this code is preparation for MGMT-7093